added GoalPageContext.tsx and refactored GoalPage to use the context …

ivonneflores45 opened the pull request:
## issue
- #23 
- the `GoalPage` used prop drilling to get it's data, now it utilizes the `GoalPageContext.tsx` to get it's data 
## changes 
- created `GoalPageContext.tsx` and migrated states from `GoalPage` to the new file.
- wrapped `GoalPage` with its `GoalPageProvider` to provide it's data
### personal issue
- i had to add `'..'` to the `vite.config.ts' to be able to run the site, but didn't push those changes

ivonneflores45 committed:
added GoalPageContext.tsx and refactored GoalPage to use the context page

Deiahri reviewed (changes requested):
While the goal page does use context now, the way it uses it defeats the purpose of the context.

Currently, the variables and functions are extracted from the context at the top level, and passed to the components. 

This is still a form of prop drilling though, which is something we try to avoid for readability.

Instead, I would advise using the context IN the components that need them.

ivonneflores45 committed:
attempt at fixing prop drilling in TaskSection

ivonneflores45 commented:
## changes
- removed `tasks`, `setTasks` from `TaskSection`
- moved `setTasks` to `TaskSection`- let me know if this is okay 
- added a typeguard to the `task` in `TaskSection`

Deiahri reviewed (approved):
Excellent! With the new context, `TaskSection` shouldn't take in any context related props.
